### PR TITLE
Use github release api to check for new cli versions

### PR DESCRIPTION
**Pull request by pomdtr**

### CODE REVIEW

File: internal/api/github.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	"github.com/google/go-github/v51/github"
+)
+
+func findRelease(matchFunc func(*github.RepositoryRelease) bool) (*github.RepositoryRelease, error) {
```
Comment (aavshr) [inline internal/api/github.go:11]: ah sorry Achille, I know I mentioned we had to problems with getting latest using api but I recall now that was in the installation script, there seems to be a direct API call to get the latest release:

https://pkg.go.dev/github.com/google/go-github/v51/github#RepositoriesService.GetLatestRelease

https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#get-the-latest-release

**Comment by pomdtr** [comment]
@aavshr what do you think of moving the script to this repo ? We could use the same strategy as for the jsonschema

**Comment by pomdtr** [comment]
> also we should update the installation scripts to also automatically get latest release from gh releases, right now the scripts need to updated and deployed manually each time

This is already done in prod.

**Review by aavshr — APPROVED** [review]
thanks @pomdtr 

> what do you think of moving the script to this repo ? We could use the same strategy as for the jsonschema

yeah better if we maintain this here.

**Pull request merged**